### PR TITLE
[ViewingRoom] Break apart components

### PR DESCRIPTION
**Pull request by damassi**
This PR breaks apart our components into more concrete chunks, also fleshes out the routing behavior better and adds two new hooks:
- `useRouter`
- `useIsRouteActive`

### CODE REVIEW

File: src/Apps/ViewingRoom/Routes/Works/index.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      <ViewingRoomCarousel />
+      <Box mt={2} mb={4}>
+        <ViewingRoomArtworkDetails />
+      </Box>
+    </Box>
+  )
+}
+
+// Top-level route needs to be exported for bundle splitting in the router
+export default WorksRoute
```
Comment (dzucconi) [inline src/Apps/ViewingRoom/Routes/Works/index.tsx:21]: 👍

File: src/Apps/ViewingRoom/Routes/Works/Components/ViewingRoomCarousel.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+  return (
+    <Flex
+      position="absolute"
+      width={50}
+      height="100%"
+      justifyContent="center"
+      alignItems="center"
+      style={{
+        backgroundColor: "#ffffff95",
```
Comment (dzucconi) [inline src/Apps/ViewingRoom/Routes/Works/Components/ViewingRoomCarousel.tsx:60]: oOoO hexalpha

File: src/Apps/ViewingRoom/Components/ViewingRoomHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+const ImageArea = styled(Box).attrs({
+  width: "50%",
+})`
+  overflow: hidden;
+`
+
+const ContentArea = styled(Flex).attrs({
+  alignItems: "center",
+  justifyContent: "center",
+  width: "50%",
```
Comment (damassi) [inline src/Apps/ViewingRoom/Components/ViewingRoomHeader.tsx:59]: Leaving as attrs as these will be responsive in next pr

File: src/Apps/ViewingRoom/Routes/Works/Components/ViewingRoomArtworkDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import React from "react"
+import { Box, Sans, Button, Serif } from "@artsy/palette"
+
+export const ViewingRoomArtworkDetails = props => {
+  return (
+    <Box width="50%" m="auto">
+      <Box>
+        <Sans size="3">Christine Sun Kim</Sans>
```
Comment (dzucconi) [inline src/Apps/ViewingRoom/Routes/Works/Components/ViewingRoomArtworkDetails.tsx:8]: Do we do like `<Sans as="h2">` or anything like that in places?

File: src/Apps/ViewingRoom/Routes/Statement/Components/ViewingRoomStatements.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+          different ways of viewing and interpreting bodies and the physical and
+          social spaces that govern our everyday lives as mutable, inclusive,
+          and open-ended. Kwak's work reimagines the form, functionality, and
+          materiality of objects in order to propose different ways of viewing
+          and interpreting bodies and the physical and social spaces that govern
+          our everyday lives as mutable, inclusive, and open-ended.
+        </Serif>
+      </Box>
+
+      <Spacer my={4} />
```
Comment (dzucconi) [inline src/Apps/ViewingRoom/Routes/Statement/Components/ViewingRoomStatements.tsx:35]: I do still think we should implement [a Stack](https://chakra-ui.com/stack) — I find this whole spacer pattern a lil weird. But maybe it's just unfamiliar. (Though I don't like all the extra DOM)

File: src/Apps/ViewingRoom/Components/ViewingRoomTabBar.tsx [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    </Flex>
+  )
+}
+
+const Tab: React.FC<LinkProps> = ({ children, to }) => {
+  const active = useIsRouteActive(to)
+
+  const borderBottom = active
+    ? `2px solid ${color("black100")}`
+    : `1px solid ${color("black10")}`
```
Comment (dzucconi) [inline src/Apps/ViewingRoom/Components/ViewingRoomTabBar.tsx:27]: Does the change in border width cause the content to shift up a bit?

File: src/Apps/ViewingRoom/Components/ViewingRoomHeader.tsx [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    <HeaderContainer>
+      <ImageArea>
+        <ResponsiveImage
+          src="https://user-images.githubusercontent.com/236943/81243255-342dcc00-8fc4-11ea-9a2b-2ab96ab67c9b.png"
+          style={{
+            backgroundSize: "cover",
+            backgroundPosition: "center",
+            height: "100%",
+          }}
+        />
```
Comment (dzucconi) [inline src/Apps/ViewingRoom/Components/ViewingRoomHeader.tsx:17]: We gotta fix the ResponsiveImage component to not use background images so we can support proper srcSet and (and it's just annoying)

Comment (dzucconi) [reply src/Apps/ViewingRoom/Components/ViewingRoomHeader.tsx:17]: [Here's the one I did recently](https://github.com/auspices/eos/blob/master/src/ResponsiveImage/ResponsiveImage.tsx)

Comment (damassi) [reply src/Apps/ViewingRoom/Components/ViewingRoomHeader.tsx:17]: Its been super super annoying through time...

File: src/Apps/ViewingRoom/Routes/Works/Components/ViewingRoomArtworkDetails.tsx (hunk 4 above)
Comment (damassi) [reply src/Apps/ViewingRoom/Routes/Works/Components/ViewingRoomArtworkDetails.tsx:8]: Ah, nice call -- we can't do `as` for "reasons" but we have an "element" prop we can use instead.

File: src/Apps/ViewingRoom/Routes/Statement/Components/ViewingRoomStatements.tsx (hunk 5 above)
Comment (damassi) [reply src/Apps/ViewingRoom/Routes/Statement/Components/ViewingRoomStatements.tsx:35]: Yeah, I used something like that [here](https://github.com/damassi/reaction/blob/27418dec416d0fe6125801609c786c9affa849e0/src/Apps/ViewingRoom/Routes/Statement/index.tsx#L15) but I think we adding Stack would improve our API a bit.

File: src/Apps/ViewingRoom/Components/ViewingRoomTabBar.tsx (hunk 6 above)
Comment (damassi) [reply src/Apps/ViewingRoom/Components/ViewingRoomTabBar.tsx:27]: Nope, not that I can see: https://staging.artsy.net/viewing-room

**Comment by artsyit** [comment]
:rocket: PR was released in [`v26.34.3`](https://github.com/artsy/reaction/releases/tag/v26.34.3) :rocket:

**Pull request merged**